subsys/usb/usb_device.c

```diff
@@ -1226,9 +1226,13 @@ static int custom_handler(struct usb_setup_packet *pSetup,
 			continue;
 		}
 
-		if ((iface->custom_handler) &&
+		/* An exception for AUDIO_CLASS is temporary and shall not be
+		 * considered as valid solution for other classes.
+		 */
+		if (iface->custom_handler &&
```

Comment:
@emob-nordic @jfischer-phytec-iot 
Mind you why not have 2 handlers here instead of the hack with `== AUDIO`?
You could have:

```
if (iface->custom_handler_all) {
   return iface->custom_handler_all(pSetup, len, data);
}
else if ((iface->custom_handler) && (if_descr->bInterfaceNumber == sys_le16_to_cpu(pSetup->wIndex)))
{
   return iface->custom_handler(pSetup, len, data);
}
```

Reply:
Well add an other handler is API addition and there are a few reasons to rework how the stack interfaces with classes and to remove something again from the API is a more complicated process as just to add something. I suggest to go with the hack but revise the interfacing after release.

Reply:
I agree with Johann. For this release let's follow this hack. I need to rethink what could be done to improve the interface between core stack and classes.